mifosng-provider/src/main/java/org/mifosplatform/portfolio/loanaccount/domain/Loan.java

```diff
@@ -531,7 +533,7 @@ public void addLoanCharge(final LoanCharge loanCharge) {
         final BigDecimal amount = calculateAmountPercentageAppliedTo(loanCharge);
         BigDecimal chargeAmt = BigDecimal.ZERO;
         BigDecimal totalChargeAmt = BigDecimal.ZERO;
-        if (loanCharge.getChargeCalculation().isPercentageBased()) {
+        if (loanCharge.getChargeCalculation().isPercentageBased() || loanCharge.getChargeCalculation().isPercentageOfDisbursementAmount()) {
```

Comment:
isPercentageOfDisbursementAmount is also percentage based. so please add this change in loanCharge.getChargeCalculation().isPercentageBased()